scripts/bump_deps.lua

```diff
@@ -1,343 +1,446 @@
-		pr_title = commit_prefix .. "bump some dependencies"
-		pr_body = "bump some dependencies"
-	else
-		relevant_commit = relevant_commit:gsub("'", "")
-		pr_title = relevant_commit
-		pr_body = relevant_commit:gsub(commit_prefix:gsub("%(", "%%("):gsub("%)", "%%)"), "")
-	end
-	pr_body = pr_body .. "\n\n(add explanations if needed)"
-	p(pr_title .. "\n" .. pr_body .. "\n")
-	create_pr(pr_title, pr_body)
+  require_executable('git')
+
+  verify_branch('deps')
+
+  local nvim_remote = find_git_remote(nil)
+  local relevant_commit = assert(run_die({
+    'git',
+    'log',
+    '--grep=' .. commit_prefix,
+    '--reverse',
+    "--format='%s'",
+    nvim_remote .. '/master..HEAD',
+    '-1',
+  }, 'Failed to fetch commits'))
+
+  local pr_title
+  local pr_body
+
+  if relevant_commit == '' then
+    pr_title = commit_prefix .. 'bump some dependencies'
+    pr_body = 'bump some dependencies'
+  else
+    relevant_commit = relevant_commit:gsub("'", '')
+    pr_title = relevant_commit
+    pr_body = relevant_commit:gsub(commit_prefix:gsub('%(', '%%('):gsub('%)', '%%)'), '')
+  end
+  pr_body = pr_body .. '\n\n(add explanations if needed)'
+  p(pr_title .. '\n' .. pr_body .. '\n')
+  create_pr(pr_title, pr_body)
+end
+
+local function usage()
+  local this_script = _G.arg[0]:match("[^/]*.lua$")
+  print(([=[
+    Bump Nvim dependencies
+
+    Usage:  nvim -l %s [options]
+        Bump to HEAD, tagged version, commit, or branch:
+            nvim -l %s --dep Luv --head
+            nvim -l %s --dep Luv --version 1.43.0-0
+            nvim -l %s --dep Luv --commit abc123
+            nvim -l %s --dep Luv --branch
+        Create a PR:
+            nvim -l %s --pr
+
+    Options:
+        -h                    show this message and exit.
+        --pr                  submit pr for bumping deps.
+        --branch <dep>        create a branch bump-<dep> from current branch.
+        --dep <dependency>    bump to a specific release or tag.
+
+    Dependency Options:
+        --version <tag>       bump to a specific release or tag.
+        --commit <hash>       bump to a specific commit.
+        --HEAD                bump to a current head.
+
+        <dependency> is one of:
+        "LuaJIT", "libuv", "Luv", "tree-sitter"
+  ]=]):format(this_script, this_script, this_script, this_script, this_script, this_script))
+end
+
+local function parseargs()
+  local args = {}
+  for i = 1, #_G.arg do
+    if _G.arg[i] == '-h' then
+      args.h = true
+    elseif _G.arg[i] == '--pr' then
+      args.pr = true
+    elseif _G.arg[i] == '--branch' then
+      args.branch = _G.arg[i+1]
+    elseif _G.arg[i] == '--dep' then
+      args.dep = _G.arg[i+1]
+    elseif _G.arg[i] == '--version' then
+      args.version = _G.arg[i+1]
+    elseif _G.arg[i] == '--commit' then
+      args.commit = _G.arg[i+1]
+    elseif _G.arg[i] == '--head' then
+      args.head = true
+    end
+  end
+  return args
 end
 
-return M
+local is_main = _G.arg[0]:match('bump_deps.lua')
```

Comment:
This is not too bad but not 100% reliable. Might still want some sort of `vim.is_main()` (?) thing that scripts can use to decide whether to run like a script or like a module. Similar to python `__name__ == "__main__"` pattern.

Reply:
Maybe something roughly like this? When neovim runs the script:
```lua
local main = loadfile('...')
_G.arg = { ... } -- set args from command line
_G.__main__ = main -- set __main__ to the main function
main()
```
And then inside the script:
```lua
if __main__ == debug.getinfo(1, 'f').func then
  ...
end
```

Reply:
that looks promising. but where/what is the `main()` function?

Reply:
`main` is the function that is loaded from `-l <file>` with `loadfile`. Right now `nlua_exec_file` does `loadfile` and then immediately executes it. It'd just need to also set that function as some variable (but only in this case obviously)

Reply:
cool idea. should `debug` be avoided though, for such common functionality? 

wonder if we can use the C API to set a magic local inside the loadfile() closure.